packages/dioxus/tests/props_spread.rs

```diff
@@ -0,0 +1,121 @@
+    pub enum FooPropsBuilder_Error_Repeated_field_open {}
+    #[doc(hidden)]
+    #[allow(dead_code, non_camel_case_types, missing_docs)]
+    impl<'a> FooPropsBuilder<'a, ((Option<&'a str>,
+                                   // pub attributes: Vec<Attribute<'a>>,
+                                  ), )> {
+        #[deprecated(note = "Repeated field open")]
+        pub fn open(self, _: FooPropsBuilder_Error_Repeated_field_open) -> FooPropsBuilder<'a, ((Option<&'a str>,
+                                                                                                 // pub attributes: Vec<Attribute<'a>>,
+                                                                                                ), )> { self }
+    }
+    #[allow(dead_code, non_camel_case_types, missing_docs)]
+    impl<'a, __open: FooPropsBuilder_Optional<Option<&'a str>>> FooPropsBuilder<'a, (__open, ), > {
+        pub fn build(self) -> FooProps<'a> {
+            let (open, ) = self.fields;
+            let open = FooPropsBuilder_Optional::into_value(open, || Default::default());
+            FooProps { open, attributes: self.attributes }
+        }
+    }
+    // -----
+
+    impl<'a, A> HasAttributesBox<'a, FooPropsBuilder<'a, (A, )>> for FooPropsBuilder<'a, (A, )> {
+        fn push_attribute(self, attr: AttributeBox<'a>) -> FooPropsBuilder<'a, (A, )> {
+            let mut attrs = Vec::from(self.attributes);
+            attrs.push(attr);
+            FooPropsBuilder { fields: self.fields, attributes: attrs, _phantom: self._phantom }
+        }
+    }
+    impl<A,> ExtendedGlobalAttributesMarker for FooPropsBuilder<'_, (A,)> {}
+    impl<A,> ExtendedAudioMarker for FooPropsBuilder<'_, (A,)> {}
+
+    use dioxus::prelude::*;
+    use dioxus_html::AudioExtension;
+
+    #[allow(non_snake_case)]
+    pub fn Foo<'a>(cx: Scope<'a, FooProps<'a>>) -> Element<'a> {
+        let muted = false;
+        let attributes = cx.props.attributes;
+        render! {
+            // rsx! {
+            //     audio {
+            //         muted: muted,
+            //     }
+            // }
+            ::dioxus::core::LazyNodes::new(move |__cx: &::dioxus::core::ScopeState| -> ::dioxus::core::VNode   {
+                static TEMPLATE: ::dioxus::core::Template = ::dioxus::core::Template { name: concat!(file!(), ":", line!(), ":", column!(), ":", "" ), roots: &[::dioxus::core::TemplateNode::Element { tag: dioxus_elements::audio::TAG_NAME, namespace: dioxus_elements::audio::NAME_SPACE, attrs: &[::dioxus::core::TemplateAttribute::Dynamic { id: 0usize }], children: &[] }], node_paths: &[], attr_paths: &[&[0u8]] };
+                let mut attrs = vec![__cx.attr(dioxus_elements::audio::muted.0, muted, dioxus_elements::audio::muted.1, dioxus_elements::audio::muted.2)];
+                for attr in attributes {
+                    attrs.push(__cx.attr(attr.name, attr.value.into_value(__cx.bump()), attr.namespace, attr.volatile));
```

Comment:
```
error[E0161]: cannot move a value of type `dyn dioxus::prelude::IntoAttributeValue<'_>`
   --> packages\dioxus\tests\props_spread.rs:101:53
    |
101 |                     attrs.push(__cx.attr(attr.name, attr.value.into_value(__cx.bump()), attr.namespace, attr.volatile));
    |                                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the size of `dyn dioxus::prelude::IntoAttributeValue<'_>` cannot be statically determined
```

I am stuck with this problem and can't solve it.
Is there a good way to create dynamic_attrs from the list of attributes that the Builder holds?

Reply:
Instead of creating a new AttributeBox type can we just use the existing Attribute struct? You will need to modify the builder to take `cx: &ScopeState` as a parameter when you first create it. Then you can use it in the `push_attribute` function to create an attribute like the attr function

Reply:
We would need to change the properties trait to take ScopeState as well which would be breaking. All of the implementers of that trait should be through the derive macro so I don't think breaking that API is an issue

Reply:
Thanks for the advice, I was able to build `Vec<Attribute<'a>>` with Builder!

Now I am having trouble with the `VNode` where I set `dynamic_attrs: &'a [Attribute<'a>]`.

```
rsx! {
    audio {
        muted: dynamic_value,.
        ... .cx.props.attributes
    }
}
```

I think I need to merge `muted` and `cx.props.attributes` in the above macro into one slice and set it to `dynamic_attrs`. I know this is possible by implementing `Copy` on `Attribute`, but I want to implement it without copying. Any ideas? I will continue to investigate bumpalo and others.

Reply:
The order of the `dynamic_attrs` array determines what element the attribute is attached to in the template ([docs](https://dioxuslabs.com/learn/0.4/contributing/walkthrough_readme#the-rsx-macro)). Because of this we need the size and order to be fixed at compile time.

We need to change [dynamic_attrs](https://github.com/DioxusLabs/dioxus/blob/master/packages/core/src/nodes.rs#L63C34-L63C34) from `dynamic_attrs: &'a [Attribute<'a>]` to `dynamic_attrs: &'a [AttributeEnum<'a>]` where `AttributeEnum` is either `SingleAttribute` or `Vec<Attribute<'a>>`.

Then when we [create](https://github.com/DioxusLabs/dioxus/blob/master/packages/core/src/create.rs#L321) and [diff](https://github.com/DioxusLabs/dioxus/blob/master/packages/core/src/diff.rs#L166) attributes, we need to handle that new case by looping over the spread attributes and setting and diffing each one individually.

Reply:
Ah, I understand now! It will take some time because of the large impact, but we will proceed with the implementation in that direction. Thanks!

Reply:
I would like to get this merged in the next few weeks to unblock some event handler changes. @atty303 is this something you are still interested in working on? If not, that is completely understandable, would you be ok with me finishing this PR?